### PR TITLE
Take SeedGeneratorFromRegionHits constructor arguments as unique_ptrs

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR is a preparatory step to change PluginFactory to return `std::unique_ptr`.

(should be the last PR before making the change on the framework side, unless I've missed something)

#### PR validation:

Code compiles.

### CODE REVIEW

File: RecoEgamma/EgammaElectronAlgos/interface/SeedFilter.h [hunk 1]
```diff
@@ -45,7 +45,7 @@ class SeedFilter {
   void seeds(edm::Event&, const edm::EventSetup&, const reco::SuperClusterRef &, TrajectorySeedCollection *);
 
  private:
-  SeedGeneratorFromRegionHits *combinatorialSeedGenerator;
+  std::unique_ptr<SeedGeneratorFromRegionHits> combinatorialSeedGenerator;
```
Comment (makortel) [inline RecoEgamma/EgammaElectronAlgos/interface/SeedFilter.h]: In retrospect this change should have been part of #26858.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/26939	7e82060
- technical 
- jenkins tests pass

**Comment by Martin-Grunewald** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**